rbc/heavydb/array.py

```diff
@@ -132,20 +126,34 @@ def T(self):
         pass
 
 
-@extending.lower_builtin(Array, types.Integer, types.StringLiteral)
-@extending.lower_builtin(Array, types.Integer, types.NumberClass)
+@extending.lower_builtin(Array, nb_types.Integer, nb_types.StringLiteral)
+@extending.lower_builtin(Array, nb_types.Integer, nb_types.NumberClass)
 def heavydb_array_constructor(context, builder, sig, args):
     return heavydb_buffer_constructor(context, builder, sig, args)
 
 
 @extending.type_callable(Array)
 def type_heavydb_array(context):
     def typer(size, dtype):
-        if isinstance(dtype, types.StringLiteral):
+        if isinstance(dtype, nb_types.StringLiteral):
             element_type = typesystem.Type.fromstring(dtype.literal_value)
-        elif isinstance(dtype, types.NumberClass):
+        elif isinstance(dtype, nb_types.NumberClass):
             element_type = typesystem.Type.fromobject(dtype)
         else:
             raise errors.NumbaNotImplementedError(repr(dtype))
         return HeavyDBArrayType((element_type,)).tonumba()
     return typer
+
+
+@extending.overload_attribute(ArrayPointer, 'ndim')
+def get_ndim(arr):
+    def impl(arr):
+        return 1
```

Comment:
Ah yes because we don't have matrix yet right?

Reply:
Yes! Arrays in heavydb are single-dimension